addons/website/models/website_snippet_filter.py

```diff
@@ -50,6 +50,13 @@ def _check_field_names(self):
                 if not field_name.strip():
                     raise ValidationError(_("Empty field name in %r") % (record.field_names))
 
+    def _get_model(self):
+        if self.filter_id:
+            return self.env[self.filter_id.model_id]
+        elif self.action_server_id:
+            return self.env[self.action_server_id.model_id.model]
+        return None
```

Comment:
in which case... It ia as well on as well the other no ? 

see: _check_data_source_is_provided constrains